spec.md

```diff
@@ -1131,6 +1131,8 @@ This RPC will be called by the CO to deprovision a volume.
 This operation MUST be idempotent.
 If a volume corresponding to the specified `volume_id` does not exist or the artifacts associated with the volume do not exist anymore, the Plugin MUST reply `0 OK`.
 
+CSI plugins SHOULD treat volumes independent from their snapshots.  If the Controller Plugin supports deleting a volume without affecting its existing snapshots, then these snapshots MUST still be fully operational and acceptable as sources for new volumes as well as appear on `ListSnapshot` calls.
```

Comment:
Thanks for the revised language, though I recommend s/MUST/SHOULD/ in the 2nd sentence. We're saying this behavior is optional, but recommended - so the use of MUST only makes it more confusing: is this required or optional behavior?

nit: one sentence per line, please.

Reply:
I forgot about the one line per sentence, sorry.

The reason for using MUST in the second sentence is that it is a conditional sentence, and I believe it to be correct.

**If** a driver does support deleting a volume independently of the snapshots then it **MUST** (not should, this is a hard requirement) allow using those snapshots normally.

If I change the MUST to SHOULD, the specs end up saying that it's OK to have a driver that allows deleting a volume with snapshots and leave its snapshots in a state that they cannot be normally used.  Which is not what we want to say.

I will add another sentence explaining the case of the controller not supporting deleting volumes independently of their snapshots to make it explicit that both options are possible, and that's why we had the `SHOULD` in the first sentence.

Thanks for the review.